backend/internal/handlers/classrooms/assignments/assignments.go

```diff
@@ -66,6 +69,80 @@ func (s *AssignmentService) createAssignment() fiber.Handler {
 	}
 }
 
+
+
+func (s *AssignmentService) acceptAssignment() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		var assignment models.AssignmentAcceptRequest
+		err := c.BodyParser(&assignment)
+		if err != nil {
+			return errs.InvalidRequestBody(models.AssignmentOutline{})
+		}
+
+		client, err := middleware.GetClient(c, s.store, s.userCfg)
+
+		if err != nil {
+			return errs.AuthenticationError()
+		}
+
+		username, err := client.GetCurrentUser(c.Context())
+		if err != nil {
+			return errs.GithubAPIError(err)
+		}
+
+		forkName := assignment.SourceRepoName + "-" + strings.ReplaceAll((username.Login), " ", "")
```

Comment:
I wonder if we should document our naming convention somewhere.

Also, I would be mindful of character limits on repository names.

Lastly, it may be nice (space permitting) to put a random hash somewhere in this to eliminate collisions if a template repo is re-used.

Reply:
For now, extracting to a helper with a TODO tag. We can codify this once we have a full assignment flow